pkg/migrations/sql/idempotent/base.sql

```diff
@@ -1429,9 +1460,55 @@ COMMENT ON FUNCTION SCHEMA_CATALOG.epoch_abort(BIGINT)
 IS 'ABORT an INSERT transaction due to the ID epoch being out of date';
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.epoch_abort TO prom_writer;
 
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.get_confirmed_unused_series(
+    series_table TEXT, potential_series_ids BIGINT[], check_time TIMESTAMPTZ
+) RETURNS BIGINT[]
+AS $func$
+DECLARE
+    r RECORD;
+BEGIN
+    FOR r IN
+        SELECT *
+        FROM SCHEMA_CATALOG.metric m
+        WHERE m.series_table = get_confirmed_unused_series.series_table
+    LOOP
+
+    --at each iteration of the loop filter potential_series_ids to only
+    --have those series ids that don't exist in the metric tables.
+    EXECUTE format(
+    $query$
+        SELECT array_agg(potential_series.series_id)
+				FROM unnest($1) as potential_series(series_id)
+				LEFT JOIN LATERAL(
+					SELECT 1
+				  FROM  %1$I.%2$I data_exists
+				  WHERE data_exists.series_id = potential_series.series_id AND time >= %3$L
```

Comment:
I don't think we should have a checktime here? That was an optimization anyway because we checked all time before previously? Semantically if we have a continuous agg before this point with data, we want to not delete the series

Reply:
Right... the problem is we do this step (marking the series for deletion) before deleting the metric chunks. I'll try changing it around and see if that will make it work without the time.